forte/utils/tagging_scheme.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2021 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utility functions related to tagging scheme.
+"""
+
+import logging
+from typing import Optional, List, Union, Tuple
+
+
+def bio_merge(
+    tags: List[str],
+    types: List[Union[str, None]],
+    index: Optional[List[Tuple[Union[int, None], Union[int, None]]]] = None,
+) -> Tuple[
+    List[Union[str, None]], List[Tuple[Union[int, None], Union[int, None]]]
+]:
+    r"""This function merged BIO-schemed augmented tagging scheme results and
+    return chunks information.
+
+    For example, BIO NER tags could be merged by passing
+    tags = ['B', 'O', 'B', 'I']
+    types = ['PER', '', 'LOC', 'LOC']
+    index = [(0, 1), (11, 19), (20, 22), (24, 26)]
+
+    After merging BIO tags, the results will be returned as
+    result_types = ['PER', 'LOC']
+    result_index = [(0, 1), (20, 26)]
+
+    The function handles 'I' with no leading 'B' tag. If we encounter
+    "I" while its type is different from the previous type, we will consider
+    this "I" as a "B" and start a new record here.
+
+    The function can also handle tags with no types, for example, in some word
+    segmentation tasks. In this case the input `types` should be set as a list
+    of None, and the returned result_type will be a list of None.
```

Comment:
Let's not return a list of `None`, instead we can return a list or a None, i.e. `Optional[List[str]]`